src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs

```diff
@@ -85,6 +92,13 @@ private void CalculateHashCode()
                     _hashValue = _hashValue * 17 + _credential.GetHashCode();
                 }
             }
+            else if (_clientCertificate != null)
+            {
+                unchecked
+                {
+                    _hashValue = _hashValue * 17 + _clientCertificate.GetHashCode();
```

Comment:
What is this?

Reply:
This is just standard way to calculate hash of objects in c# 
since SqlConnectionPool is extended to have ClientCertificate and have cached _hash value which is used for comparison of sqlconnectionpool objects.
So i just extended hash to also save certificate hash code